### DISCUSSION TITLE
Post-Process Functions

### DISCUSSION THREAD

**Discussion by OBWANDO** [o3de/rfcs#16] [Ideas]
Enzo0207 posted:
We see that UE4 has the following postprocess functions:
- [ ] Bloom Convolution
- [ ] Bloom Dirt Mask
- [x] Ambient Cubemap
    - Ambient Cubemap type functionality in o3de currently can be accomplished with the 'Global SkyLight (IBL) Component'. Additionally, the 'Reflection Probe Component' can load a custom specular reflection image (also from hdri).
- [ ] Panini Projection
- [ ] PostProcess Materials
- [ ] Lens Flare
- [ ] Scene Fringe
- [ ] Motion Blur
    - We have a Motion Vectors pass implemented on the rendering side but do not yet have a Motion Blur pass that uses it. This work just needs to get prioritized on the short term roadmap.
- [ ] Screen Space Reflection
- [ ] Vignette
Will these functions be further developed in O3DE?

It may be good to get some thoughts from the community on how some of these may be important and should we look at putting together a RFC for the Presentation SIG to consider for the future roadmap.

This would be a great thread for us to discuss and also introduce any other functions that may be useful.

**Comment by HogJonny-AMZN** [comment]
Ambient Cubemap type functionality in o3de currently can be accomplished with the 'Global SkyLight (IBL) Component'.  Additionally, the 'Reflection Probe Component' can load a custom specular reflection image (also from hdri).

**Comment by HogJonny-AMZN** [comment]
We have a Motion Vectors pass implemented on the rendering side but do not yet have a Motion Blur pass that uses it. This work just needs to get prioritized on the short term roadmap.

**Comment by HogJonny-AMZN** [comment]
Screen Space Reflections have been implemented in the rendering there is a sample in the AtomSampleViewer (ASV).  This just needs to be exposed to the engine/editor as a component, this work I believe will be happening soon.

**Comment by HogJonny-AMZN** [comment]
We have planned the work for a 'Color Grading / Color Adjustment' PostFX pass and component, that work will start soon as well.  There is already a Look Modification component that loads a pre-baked LUT but the workflow to generate the lut files is clunky.

**Comment by Enzo0207** [comment]
Q1. bloom convolution is not supported now in O3DE, will you consider supporting it in the future?
Q2. Bloom Tint bug. as shown in pictures, we set tint0 to red, set tint1 to green in Both UE4 and O3DE. In UE4, the inner circle is red and the outer circle is green, but in O3DE, the all bloom effect is red. Will it be resolved in the future?

As we cannot upload the pictures of Q2, please see them on the SIG-Presentation.

**Comment by Enzo0207** [comment]
And also, we have read your document 'AtomRenderer-Features' section in which you plan two types' AA:
MSAA: enabled by default
SMAA: can be enabled SMAAConfiguration.azasset()
we wonder that:
Q1. Are these features already implemented? How to switch on/off them like using IDE/console?
Q2. Is there any other AA algorithms developed in O3DE? For example, we find specular AA in material editor. If possible, we do wanna a AA list...  
Thx !

**Comment by Enzo0207** [comment]
In addition, how does PostFx Gradient Weight Modifier influence weight of PostFxLayer? It seems using camera position and gradient entity?

**Comment by Enzo0207** [comment]
how to adjust resolution for rendering in O3DE？some config files or some certain buttons in editor？

**Comment by huawei-shiyue** [comment]
Hi OBWANDO, I meet some problems when customizing new upsampling pass in PostProcess block. When it creates new color attachment with multipliers larger than one, the rest passes sample wrong with other inputs like depth, which still keep original size. 
Is there any good suggestion to support subsequent passes which require other inputs like depth?

**Comment by thefranke** [comment]
The volumetric cloud & fog request seems related (the fog part, not the clouds):
https://github.com/o3de/rfcs/discussions/21
